### ISSUE TITLE
Inconsistent behavior of create_user method

### ISSUE THREAD

**Issue by 8byr0** [Corvia/django-tenant-users#202]
Hi there! 
I was testing login feature of an app using this package and was quite surprised to notice that it was possible for a user to register as much as he wants as long as his profile is not active yet. 

```python
with schema_context("public"):
  user: PlatformUser = PlatformUser.objects.create_user(
        email=email,
        password=password,
        first_name=first_name,
        last_name=last_name,
        is_active=True,
  )
```
I was expecting this call to fail if ran twice with the same input but it did not so I checked the code of `create_user` method and saw this:
https://github.com/Corvia/django-tenant-users/blob/07cb5720d2ce638da05ca1730b83cac203a08fd7/tenant_users/tenants/models.py#L269-L270

I do not really understand why you check on condition `profile.is_active` (line 269). I mean... Even if the profile is not active the user still exists and is created!

I'd suggest to remove this assertion or to make it optional from settings. Would it make sense?

I can submit a PR for that if you think it's relevant.

Cheers

**Comment by 8byr0** [comment]
For anyone having the same issue, here's a simple workaround:

```python
from tenant_users.tenants.models import ExistsError
from django_tenants.utils import schema_context, get_public_schema_name

try:
    # Check if an account with this email exists
    # Reason: https://github.com/Corvia/django-tenant-users/issues/202
    MyUserModel.objects.get(email=email)
    raise ExistsError("User already exists!")

except MyUserModel.DoesNotExist:
    pass
try:
    with schema_context(get_public_schema_name()):
        user: MyUserModel = MyUserModel.objects.create_user(
            email=email,
            password=password,
            first_name=first_name,
            last_name=last_name,
            is_active=True,
        )
except Exception as err:
    # create_user does not only create user, it does other things
    # after it that may fail
    # If it fails, we must delete any created user during
    # the process
    try:
        MyUserModel.objects.get(email=email).delete(force_drop=True)
    except MyUserModel.DoesNotExist:
        pass

    raise Exception from err
```

**Comment by 8byr0** [comment]
I will make a PR for that yes.

Another thing is the public tenant; You cannot use this package if no public tenant exists. 
What I ended up doing is a custom command like this:
```python
from django.conf import settings
from django.core.management.base import BaseCommand
from django_tenants.utils import get_public_schema_name, schema_context
from tenant_users.tenants.utils import create_public_tenant

from mytenants.models import MyTenantModel


class Command(BaseCommand):
    help = "Create the public tenant if it does not exist yet"

    def handle(self, *args, **options):
        with schema_context(get_public_schema_name()):
            try:
                MyTenantModel.objects.get(schema_name=get_public_schema_name()).schema_name
                self.stdout.write(self.style.NOTICE("Public tenant already exists"))
            except MyTenantModel.DoesNotExist:
                create_public_tenant(settings.BASE_DOMAIN, "admin@domain.com")
                self.stdout.write(
                    self.style.SUCCESS("Successfully created public tenant")
                )
```

This works fine cause I can call it at runtime after my migrations but since `create_public_tenant` manages the tenant owner creation, it would be better to create it automatically if it does not exist instead of raising an error because it could not be found.

I could imagine something like this:
```python 
# server/settings.py
CREATE_PUBLIC_TENANT_AT_RUNTIME = True
PUBLIC_TENANT_OWNER_EMAIL = "john.doe@domain.com"
PUBLIC_TENANT_DOMAIN = "domain.com"
```

It would let developers choose between:
- Use this default settings
- Disable auto creation and keep actual behavior (manually call `create_public_tenant`)

**Comment by Dresdn** [comment]
Are you saying that upon initial project standup, auto-creating the public schema would be helpful?

**Comment by 8byr0** [comment]
Yes, that's what I'm saying. I'd be curious to have feedback from others on this ?
________________________________

**Comment by Dresdn** [comment]
The conversation is probably starting to stray a bit, but overall I think that having a management command to provision would be great. There was a [management command in the old "test_project"](https://github.com/Corvia/django-tenant-users/blob/0.4.0/dtu_test_project/core/management/commands/setup_dtu_tenants.py) that did exactly that, so maybe parameterizing it and putting it into the project would be beneficial.

To close off your other thoughts, personally, I'm not really a fan of things happening "automatically" with the use of one-time-use settings.

What do you think?

**Comment by 8byr0** [comment]
You're right we should move this discussion to another issue.

I agree on the point that things done automagically are not the best for users. On the other hand, this particular case `create_public_tenant` is required to be done before starting the server / at runtime and will fail if not done so users have to do it. 
My proposal is just to extend the scope of ways to create the public tenant:
- Let the user manage the creation (default)
- Let dtu manage it it based on settings values (runtime)
- Run command after migrations (pre-runtime)


Let's move that to another issue

**Comment by Dresdn** [comment]
@8byr0 - I had some time and finally sat and took a look at this. Knowing what I know now, I would have initially responded to you differently.

When thinking about a "single-tenant" project, your initial assessment is 100% correct: it'd be strange to be able to run `create_user` twice. However, in a multi-tenant project, there is a method to the madness. Here's a use case to describe:

a. First time user registers at red.tenant.com. Profile is created and set to active.
b. Said user leaves, but to preserve history, the profile is set to inactive.
c. Later, user now joins blue.tenant.com. Since the profile exists, `create_user` uses the original profile, but just toggles the profile to be active, allowing the user to login.

This issue ultimately could be a "documentation needed" to explain the current behavior. At the end of the day, this really just allows an admin to set a user to inactive, but then if they register for any tenant, the profile will get re-used, so I can't think of a downside of changing the code.

Looking at your workaround, you're really just duplicating what already exists as you cannot create multiple instances with `create_user()` - we check for an existing profile, but ultimately only end up just re-activating it if it's disabled.

Let me know your thoughts, as I can add this to the list of things needed in the documentation.

**Comment by Dresdn** [comment]
Closing as https://django-tenant-users.readthedocs.io/en/latest/pages/concepts.html addresses this.
